### PR TITLE
Availability test

### PR DESCRIPTION
**Pull request by JackyWYX**
* Added unit test to the module `staking/availabiltiy`

## Issue

Solved issue: [#2350](https://github.com/harmony-one/harmony/issues/2350)

### Unit Test Coverage

Before:

```
PASS
coverage: 0.0% of statements
ok      github.com/harmony-one/harmony/staking/availability     0.352s
```

After:

```
PASS
coverage: 88.4% of statements
ok      github.com/harmony-one/harmony/staking/availability     0.304s
```

### CODE REVIEW

File: staking/availability/interface.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+// Reader ..
+type Reader interface {
+	ReadValidatorSnapshot(
+		addr common.Address,
+	) (*staking.ValidatorWrapper, error)
+}
+
+// Header is the interface of block.Header for calculating the BallotResult.
+type Header interface {
```
Comment (fxfactorial) [inline staking/availability/interface.go:18]: `RoundReader`

File: staking/availability/interface.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	Number() *big.Int
+	ShardID() uint32
+	LastCommitBitmap() []byte
+}
+
+// StateDB is the interface of state.DB
+type StateDB interface {
```
Comment (fxfactorial) [inline staking/availability/interface.go:25]: ValidatorState

Comment (jwangdx) [reply staking/availability/interface.go]: Thx for naming recommendation. Fixed.

Comment (JackyW47) [reply staking/availability/interface.go]: Thanks for naming recommendation. Fixed.

File: staking/availability/measure_test.go [hunk 3]
```diff
@@ -1,28 +1,736 @@
...
+		msk := byte(1) << uint(idx&7)
+		res[byt] ^= msk
+	}
+	return res, nil
 }
 
-func TestCompute(t *testing.T) {
-	t.Log("Unimplemented")
+func makeTestWrapper(addr common.Address, numSigned, numToSign int64) staking.ValidatorWrapper {
+	var vm staking.ValidatorWrapper
```
[the remark this answers is not in the document]
Comment (JackyW47) [reply staking/availability/measure_test.go:731]: Thanks for naming recommendation. Fixed.

File: staking/availability/interface.go (hunk 1 above)
Comment (JackyW47) [reply staking/availability/interface.go]: Thanks for naming recommendation. Fixed.

**Comment by fxfactorial** [comment]
Jenkins, test this please

**Comment by rlan35** [comment]
Awesome job!

**Pull request merged**